### PR TITLE
Refactored Indicator1 disaggregated by entry points

### PR DESCRIPTION
**Pull request by tresorluz**

### CODE REVIEW

File: report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/test_utils/models/ObsValueTypeEnum.java [hunk 1]
```diff
@@ -4,7 +4,8 @@ public enum ObsValueTypeEnum {
     CODED("value_coded"),
     DATE_TIME("value_datetime"),
     TEXT("value_text"),
-    NUMERIC("value_numeric");
+    NUMERIC("value_numeric"),
+	BOOLEAN("value_boolean");
```
Comment (tmvumbi2) [inline report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/test_utils/models/ObsValueTypeEnum.java]: @tresorluz this enum isn't being used, it can be safely removed

File: metadata/reportssql/hivtmi_report_functions.sql [hunk 2]
```diff
@@ -101,59 +75,9 @@ SELECT
 FROM
   patient pat
 WHERE
-  getObsDatetimeValueInSection(pat.patient_id, "c6c08cdc-18dc-4f42-809c-959621bc9a6c", "b70dfca0-db21-4533-8c08-4626ff0de265") BETWEEN p_startDate AND p_endDate AND
+  getObsDatetimeValueInSection(pat.patient_id, "c6c08cdc-18dc-4f42-809c-959621bc9a6c", "d6cc3709-ffa0-42eb-b388-d7def4df30cf") BETWEEN p_startDate AND p_endDate AND
```
Comment (tmvumbi2) [inline metadata/reportssql/hivtmi_report_functions.sql]: @tresorluz why did you change the parent concept (i.e. the second uuid parameter) here? The new concept uuid used refers to "At ANC Enrolment" which is not the right parent concept for HIV Test Date. The old uuid value (i.e. "b70dfca0-db21-4533-8c08-4626ff0de265") was the right one, it refers to the sub-section "HIV Test".

File: report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        testDataGenerator.program.enrollPatientIntoHIVProgram(
+			patientId,
+			new LocalDate(2022, 11, 3),
+			ConceptEnum.WHO_STAGE_1,
+			TherapeuticLineEnum.FIRST_LINE,
+			new LocalDate(2022, 11, 3)
+		);
+
+        testDataGenerator.ancInitialForm.recordAtAncEnrolmentHivTestDateAndResult(
```
Comment (tmvumbi2) [inline report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java]: @tresorluz we shouldn't need to update the ANC form. We rather need to populate the HIV test date

File: report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+import org.joda.time.LocalDate;
+import org.joda.time.LocalDateTime;
+import org.junit.Test;
+
+
+public class HivtmiIndicator1ByEntrypointsTests extends BaseReportTest {
+    @Test
+    public void shouldCountHivPatientByEntryPoints() throws Exception {
+		int patientId = testDataGenerator.registration.createPatient(
```
Comment (tmvumbi2) [inline report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java]: @tresorluz could you please fix the indentation, and use 4 spaces for each indentation?

File: report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/test_utils/HivTestingAndCounsellingFormDataGenerator.java [hunk 5]
```diff
@@ -39,6 +39,14 @@ public int recordHTCResult(int patientId, LocalDateTime obsDateTime, ConceptEnum
         return TestDataGenerator.recordFormCodedValue(patientId, obsDateTime, conceptTree, result, encounterId, stmt);
     }
 
+    public int recordResultReceived(int patientId, LocalDateTime obsDateTime, boolean result, Integer encounterId) throws Exception {
+		List<ConceptEnum> conceptTree= new ArrayList<ConceptEnum>();
```
Comment (tmvumbi2) [inline report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/test_utils/HivTestingAndCounsellingFormDataGenerator.java:51]: @tresorluz could you please fix the indentation here as well?

File: report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import org.jembi.bahmni.report_testing.test_utils.models.VisitTypeEnum;
+import org.jembi.bahmni.report_testing.test_utils.models.ReportEnum;
+import org.jembi.bahmni.report_testing.test_utils.models.GenderEnum;
+
+import org.joda.time.LocalDate;
+import org.joda.time.LocalDateTime;
+import org.junit.Test;
+
+
+public class HivtmiIndicator1ByEntrypointsTests extends BaseReportTest {
```
Comment (tmvumbi2) [inline report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java:19]: @tresorluz we need to add unit tests to test each entry point individually. We also need unit tests for edge cases, like for instance when
- no entry point has been selected
- HIV test date is outside the reporting period
- no result has been captured

**Review by tmvumbi2 — CHANGES_REQUESTED** [review]

File: metadata/reportssql/hivtmi_report_functions.sql (hunk 2 above)
Comment (tresorluz) [reply metadata/reportssql/hivtmi_report_functions.sql]: We made this change with you because this concept b70dfca0-db21-4533-8c08-4626ff0de265 is in hiv testing and counseling form not in ANC initial form

File: report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java [hunk 7]
```diff
@@ -17,73 +16,262 @@
...
+                encounterId);
 
+        // execute
         String query = readReportQuery(ReportEnum.CDV_TS_REPORT,
-         "indicator1_CDV_TS_number_of_people_tested_in_the_month_disaggregation_entry_point.sql",
-          new LocalDate(2022, 11, 1),
-          new LocalDate(2022, 11, 30)
-          );
+                "indicator1_CDV_TS_number_of_people_tested_in_the_month_disaggregation_entry_point.sql",
+                new LocalDate(2023, 01, 1),
```
Comment (tmvumbi2) [inline report-testing-framework/report-testing/src/test/java/org/jembi/bahmni/report_testing/cdv_ts_reports/HivtmiIndicator1ByEntrypointsTests.java]: @tresorluz for this test, the HIV test date needs to be within the reporting period (i.e. we need to use nov 2022 here)

**Comment by tmvumbi2** [comment]
@tresorluz the current branch has a conflict with the dev branch (the conflicted file is backup-openmrs.sql). Could you please merge the latest version of the dev branch into this branch on your local environment, and then resolve the conflict?

**Review by tmvumbi2 — CHANGES_REQUESTED** [review]

**Review by tmvumbi2 — APPROVED** [review]
